mir_connector/inorbit_mir_connector/src/connector.py

```diff
@@ -271,3 +290,122 @@ def _execution_loop(self):
+            "distance_threshold": MIR_MOVE_DISTANCE_THRESHOLD,
+        }
+        if connector_type == "MiR100" and firmware_version == "v2":
+            param_values["retries"] = 5
+        elif connector_type == "MiR250" and firmware_version == "v3":
+            param_values["blocked_path_timeout"] = 60.0
+        else:
+            self._logger.warning(
+                f"Not supported connector type and firmware version combination for waypoint "
+                f"navigation: {connector_type} {firmware_version}. Will attempt to send waypoint "
+                "based on firmware version."
+            )
+            if firmware_version == "v2":
+                param_values["retries"] = 5
+            else:
+                param_values["blocked_path_timeout"] = 60.0
+
+        action_parameters = [
+            {"value": v, "input_name": None, "guid": str(uuid.uuid4()), "id": k}
+            for k, v in param_values.items()
+        ]
+        self.mir_api.add_action_to_mission(
+            action_type="move_to_position",
+            mission_id=mission_id,
+            parameters=action_parameters,
+            priority=1,
+        )
+        self.mir_api.queue_mission(mission_id)
+
+    def setup_connector_missions(self):
+        """Find and store the required missions and mission groups, or create them if they don't
+        exist."""
+        self._logger.info("Setting up connector missions")
+        # Find or create the missions group
+        mission_groups: list[dict] = self.mir_api.get_mission_groups()
+        group = next(
+            (x for x in mission_groups if x["name"] == MIR_INORBIT_MISSIONS_GROUP_NAME), None
+        )
+        self.tmp_missions_group_id = group["guid"] if group is not None else str(uuid.uuid4())
+        if group is None:
+            self._logger.info(f"Creating mission group '{MIR_INORBIT_MISSIONS_GROUP_NAME}'")
+            group = self.mir_api.create_mission_group(
+                feature=".",
+                icon=".",
+                name=MIR_INORBIT_MISSIONS_GROUP_NAME,
+                priority=0,
+                guid=self.tmp_missions_group_id,
+            )
+            self._logger.info(f"Mission group created with guid '{self.tmp_missions_group_id}'")
+        else:
+            self._logger.info(
+                f"Found mission group '{MIR_INORBIT_MISSIONS_GROUP_NAME}' with "
+                f"guid '{self.tmp_missions_group_id}'"
+            )
+
+    def cleanup_connector_missions(self):
+        """Delete the missions group created at startup"""
+        self._logger.info("Cleaning up connector missions")
+        self._logger.info(f"Deleting missions group {self.tmp_missions_group_id}")
+        self.mir_api.delete_mission_group(self.tmp_missions_group_id)
+
+    def start_missions_garbage_collector(self):
+        """Start the garbage collector for missions"""
+        signal.signal(signal.SIGALRM, self._missions_gc_callback)
```

Comment:
Hey I'm not sure SIGALRM is intended for this. In fact it's super limited since you can't have multiple alarms for different things. Also the garbage collector would run in the main thread, right ?

Perhaps instead we could launch a new thread that runs GC in a loop waiting `MISSIONS_GARBAGE_COLLECTION_INTERVAL_SECS` after each iteration. What do you think ?

Reply:
You are right. I replaced that method with running the gc as a new process. Using threads was not very suitable because they cannot be terminated, and using flags to stop the loop in a 12h sleep didn't make sense when shutting down the connector.